.github/workflows/basic.yml

```diff
@@ -80,80 +80,84 @@ jobs:
     name: Ubuntu
     strategy:
       matrix:
-        os: ['ubuntu-20.04', 'ubuntu-22.04']
+        os_version: ['20.04', '22.04']
         build_type: [Debug, Release]
         compiler: [{c: gcc, cxx: g++}]
         pool_tracking: ['ON', 'OFF']
         shared_library: ['OFF']
         os_provider: ['ON']
         level_zero_provider: ['ON']
         include:
-          - os: 'ubuntu-20.04'
+          - os_version: '20.04'
             build_type: Release
             compiler: {c: gcc-7, cxx: g++-7}
             shared_library: 'OFF'
             os_provider: 'ON'
             pool_tracking: 'ON'
             level_zero_provider: 'ON'
-          - os: 'ubuntu-22.04'
+          - os_version: '22.04'
             build_type: Release
             compiler: {c: clang, cxx: clang++}
             shared_library: 'OFF'
             os_provider: 'ON'
             pool_tracking: 'ON'
             level_zero_provider: 'ON'
-          - os: 'ubuntu-22.04'
+          - os_version: '22.04'
             build_type: Release
             compiler: {c: gcc, cxx: g++}
             shared_library: 'ON'
             os_provider: 'ON'
             pool_tracking: 'ON'
             level_zero_provider: 'ON'
-          - os: 'ubuntu-22.04'
+          - os_version: '22.04'
             build_type: Debug
             compiler: {c: gcc, cxx: g++}
             shared_library: 'ON'
             os_provider: 'ON'
             pool_tracking: 'ON'
             level_zero_provider: 'ON'
           # test os_provider='OFF' with shared_library='ON'
-          - os: 'ubuntu-22.04'
+          - os_version: '22.04'
             build_type: Release
             compiler: {c: gcc, cxx: g++}
             shared_library: 'ON'
             os_provider: 'OFF'
             pool_tracking: 'ON'
             level_zero_provider: 'ON'
           # test os_provider='OFF' with shared_library='OFF'
-          - os: 'ubuntu-22.04'
+          - os_version: '22.04'
             build_type: Release
             compiler: {c: gcc, cxx: g++}
             shared_library: 'OFF'
             os_provider: 'OFF'
             pool_tracking: 'ON'
             level_zero_provider: 'ON'
           # test level_zero_provider='OFF'
-          - os: 'ubuntu-22.04'
+          - os_version: '22.04'
             build_type: Release
             compiler: {c: gcc, cxx: g++}
             shared_library: 'OFF'
             os_provider: 'ON'
             pool_tracking: 'ON'
             level_zero_provider: 'OFF'
-    runs-on: ${{matrix.os}}
+    runs-on: ubuntu-${{matrix.os_version}}
+    container:
+      image: ubuntu:${{matrix.os_version}}
```

Comment:
we should use pinned version (due to security score), so we have to use hash version instead of a "nice" number

and at best we should use the full link to the image (e.g. `registry.hub.docker.com/library/ubuntu`)